psl/psl-core/src/datamodel_connector.rs

```diff
@@ -73,6 +73,9 @@ pub trait Connector: Send + Sync {
     /// The referential actions supported by the connector.
     fn referential_actions(&self, relation_mode: &RelationMode) -> BitFlags<ReferentialAction>;
 
+    /// The referential actions supported when using relationMode = "prisma" by the connector.
+    fn emulated_referential_actions(&self, relation_mode: &RelationMode) -> BitFlags<ReferentialAction>;
```

Comment:
Can we add a note in the doc string explaining why we don't want to emulate some of the referential actions only on some of the connectors when `relationMode = "prisma"`?

Reply:
```suggestion
    /// The referential actions supported when using relationMode = "prisma" by the connector.
    /// There are in fact scenarios in which the set of emulated referential actions supported may change
    /// depending on the connector. For example, Postgres' NoAction mode behaves similarly to Restrict
    /// (raising an error if any referencing rows still exist when the constraint is checked), but with
    /// a subtle twist we decided not to emulate: NO ACTION allows the check to be deferred until later
    /// in the transaction, whereas RESTRICT does not.
    fn emulated_referential_actions(&self, relation_mode: &RelationMode) -> BitFlags<ReferentialAction>;
```